scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,580 @@
+    [[ -z "$context" ]] && continue
+    [[ "$context" == "$(kubectl config current-context || true)" ]] && cc="*" || cc=""
+    if kubectl --context $context get secrets -A > /dev/null 2>&1 ; then
+        echov -e "${COLOR_GREEN}\xE2\x9C\x94${COLOR_NONE} ${COLOR_INFO}${context}${COLOR_NONE} ${COLOR_GREEN}$cc${COLOR_NONE}"
+        [[ -z "$cc" ]] && valid_contexts+=("$context") || valid_contexts=("$context" "${valid_contexts[@]}")
+    else
+        echov -e "${COLOR_RED}X${COLOR_NONE} ${COLOR_INFO}${context}${COLOR_NONE} ${COLOR_GREEN}$cc${COLOR_NONE}"
+    fi
+done
+
+if [[ -z "${valid_contexts[@]}" ]] ; then
+    echoerr "No valid context(s) found in the kubeconfig file $KUBECONFIG!"
+    exit 1
+fi
+
+
+###############################################################################
+# Look for KubeStellar Helm chart
+###############################################################################
+echotitle "KubeStellar:"
+helm_context=""
+for context in "${valid_contexts[@]}" ; do
+    HELM_LIST="$TMPFOLDER/helm-list.txt"
+    helm --kube-context $context list --no-headers -A 2> /dev/null > "$HELM_LIST" || true
+    while IFS= read -r line; do
+        [[ -z "$line" ]] && continue
+        name="$(echo "$line" | awk '{print $1}')"
+        namespace="$(echo "$line" | awk '{print $2}')"
+        version="$(echo "$line" | awk '{print $NF}')"
+        secret="$(kubectl --context $context get secret -n "$namespace" -l "owner=helm,name=$name" --no-headers -o name 2> /dev/null || true)"
+        secret="${secret##*/}"
+        if [[ ! -z "$(kubectl --context $context get secret $secret -n "$namespace" -o jsonpath='{.data.release}' | base64 -d | base64 -d | gzip -d | jq -r '.chart.metadata.description' | grep -i "KubeStellar" 2> /dev/null || true)" ]] ; then
+            helm_context="$context"
+            echo -e "- Helm chart ${COLOR_INFO}${name}${COLOR_NONE} (${COLOR_INFO}v${version}${COLOR_NONE}) in namespace ${COLOR_INFO}${namespace}${COLOR_NONE} in context ${COLOR_INFO}${context}${COLOR_NONE}"
+            echo -e "  - Secret=${COLOR_INFO}${secret}${COLOR_NONE} in namespace ${COLOR_INFO}${namespace}${COLOR_NONE}"
+            if [[ "$arg_yaml" == "true" ]] ; then
+                mkdir -p "$OUTPUT_FOLDER/kubestellar-core-chart"
+                kubectl --context $helm_context get secret $secret -o jsonpath='{.data.release}' | base64 -d | base64 -d | gzip -d | jq -r '.info'      > "$OUTPUT_FOLDER/kubestellar-core-chart/info.json"
+                kubectl --context $helm_context get secret $secret -o jsonpath='{.data.release}' | base64 -d | base64 -d | gzip -d | jq -r '.chart'     > "$OUTPUT_FOLDER/kubestellar-core-chart/chart.json"
+                kubectl --context $helm_context get secret $secret -o jsonpath='{.data.release}' | base64 -d | base64 -d | gzip -d | jq -r '.manifest'  > "$OUTPUT_FOLDER/kubestellar-core-chart/manifest.yaml"
+            fi
+            break
+        fi
+    done < "$HELM_LIST"
```

Comment:
This could be `<<< $helm_list` (take stdin from the content of a shell variable) rather than `< $HELM_LIST`, saving the trouble to define a special filename (in addition to the shell variable).

Reply:
ok, I started to be worried about what works in old bashes